src/bump/type.go

```diff
@@ -53,9 +59,9 @@ func From(previous, current *semver.Version) Type {
 }
 
 // Bump returns a new version after bumping it according to the specified bump bump.
-func Bump(version *semver.Version, bt Type) *semver.Version {
+func Bump(version *semver.Version, bt Type) (*semver.Version, error) {
```

Comment:
I agree with the goal of make `next-version` fail, but I think this is a responsibility of the command. Could we make the command fail if `version == Bump(version, None)`, and not return any error here?

Reply:
I've moved this error logic in the last commit to the bumper.BumpSource, this method is already erroring when there are no tags, and I find it correct, because you are trying to bump but the source doesn't have anything to bump. 
If you still think this is better responsibility of the command, then this :
```
	sort.Slice(versions, func(i, j int) bool {
		// We use GreaterThan as lessFunc to sort from largest to smallest.
		return versions[i].GreaterThan(versions[j])
	})
```
needs to be duplicated in the command in order to know what is the current version in the source and then make the comparison.